src/routes/safe/components/Settings/ManageOwners/AddOwnerModal/index.tsx

```diff
@@ -25,12 +26,13 @@ export type OwnerValues = {
 export const sendAddOwner = async (
   values: OwnerValues,
   safeAddress: string,
-  safeVersion: string,
   txParameters: TxParameters,
   dispatch: Dispatch,
+  connectedWalletAddress: string,
 ): Promise<void> => {
-  const gnosisSafe = getGnosisSafeInstanceAt(safeAddress, safeVersion)
-  const txData = gnosisSafe.methods.addOwnerWithThreshold(values.ownerAddress, values.threshold).encodeABI()
+  const sdk = await getSafeSDK(connectedWalletAddress, safeAddress)
+  const safeTx = await sdk.getAddOwnerTx(values.ownerAddress, +values.threshold)
+  const txData = safeTx.data.data
```

Comment:
Have you tested this? Does `txData` match that required by `createTransaction()`?

Reply:
Yes and I also checked with @germartinez that I quote here
> txData is always a string (something like: 0x...). sdk.getAddOwnerTx() returns an object (let's call it safeTx) of type SafeTransaction (https://github.com/gnosis/safe-core-sdk/blob/faf64592c24e1656c18c16cc8ec78977ad0a89c9/packages/safe-core-sdk-types/src/types.ts#L39).  Use the property data safeTx.data to get the transaction content of type SafeTransactionData (https://github.com/gnosis/safe-core-sdk/blob/faf64592c24e1656c18c16cc8ec78977ad0a89c9/packages/safe-core-sdk-types/src/types.ts#L13). This extends the interface MetaTransactionData, where you can find the data of the AddOwnerTx